pytensor/tensor/rewriting/linalg.py

```diff
@@ -377,3 +381,26 @@ def local_lift_through_linalg(
                 return [block_diag(*inner_matrices)]
             else:
                 raise NotImplementedError  # pragma: no cover
+
+
+@register_canonicalize
+@register_stabilize
+@register_specialize
+@node_rewriter([SVD])
+def local_svd_uv_simplify(fgraph, node):
+    """If we have more than one `SVD` `Op`s and at least one has keyword argument
+    `compute_uv=True`, then we can change `compute_uv = False` to `True` everywhere
+    and allow `pytensor` to re-use the decomposition outputs instead of recomputing.
+    """
+    (x,) = node.inputs
+    compute_uv = False
+
+    for cl, _ in fgraph.clients[x]:
```

Comment:
You have to be careful because if the output of the SVD is an output of the function one of the clients will be a string `"output"` and the call `cl.op` will fail.